packages/frontend/src/features/notification/stores/useNotificationStore.tsx

```diff
@@ -0,0 +1,69 @@
+import { create } from 'zustand'
+import { NotificationAction } from '../types/NotificationTypes'
+
+interface Notification {
+    id: number
+    type: string
+    message: string
+    time: string
+    isRead: boolean
+    actions?: NotificationAction[]
+    targetId?: string
+}
+
+interface NotificationStore {
+    notifications: Notification[]
+    showNotificationDot: boolean
+    addNotification: (notification: Notification) => void
+    markAsRead: (id: number) => void
+    loadNotifications: () => void
+    reset: () => void
+    clearNotificationDot: () => void
+}
+
+// Define initial state
+const initialState = {
+    notifications: [] as Notification[],
+    showNotificationDot: false,
+}
+
+export const useNotificationStore = create<NotificationStore>((set) => ({
+    ...initialState,
+    addNotification: (notification) =>
+        set((state) => {
+            const updatedNotifications = [...state.notifications, notification]
+            localStorage.setItem(
+                'notifications',
+                JSON.stringify(updatedNotifications),
+            )
+            return {
+                notifications: updatedNotifications,
+                showNotificationDot: true,
+            }
+        }),
+    markAsRead: (id) =>
+        set((state) => {
+            const updatedNotifications = state.notifications.map((n) =>
+                n.id === id ? { ...n, isRead: true } : n,
+            )
+            localStorage.setItem(
+                'notifications',
+                JSON.stringify(updatedNotifications),
+            )
+            return { notifications: updatedNotifications }
+        }),
+    loadNotifications: () => {
+        const storedNotifications = localStorage.getItem('notifications')
+        if (storedNotifications) {
+            set({ notifications: JSON.parse(storedNotifications) })
+        }
+    },
+    reset: () => {
+        localStorage.removeItem('notifications')
+        set({ ...initialState })
+    },
+    clearNotificationDot: () => set({ showNotificationDot: false }),
+}))
+
+// Load stored notifications upon initializing
+useNotificationStore.getState().loadNotifications()
```

Comment:
Following React's best practices, should not use hooks outside of components.

Reply:
Since I already called it inside a component, so i will just delete this line.